tests/unit/test_operator.py

```diff
@@ -1,482 +1,398 @@
-    )
 
-    # Spot check for seldon init-container credentials
-    assert (
-        b64decode(
-            secrets_dict[f"{charm_name}-seldon-init-container-s3-credentials"]["data"][
-                "RCLONE_CONFIG_S3_ACCESS_KEY_ID"
-            ]
-        ).decode("utf-8")
-        == os_data_dict["access-key"]
-    )
-    assert len(secrets_dict[f"{charm_name}-seldon-init-container-s3-credentials"]["data"]) == 6
-    assert (
-        b64decode(
-            secrets_dict[f"{charm_name}-seldon-init-container-s3-credentials"]["data"][
-                "RCLONE_CONFIG_S3_ENDPOINT"
-            ]
-        ).decode("utf-8")
-        == f"http://{os_data_dict['service']}.{os_data_dict['namespace']}:{os_data_dict['port']}"
-    )
-
-    # Confirm default_artifact_root config
-    args = pod_spec[0]["containers"][0]["args"]
-    default_artifact_root_arg_index = args.index("--default-artifact-root")
-    expected_bucket_name = f"s3://{default_artifact_root}/"
-    actual_bucket_name = args[default_artifact_root_arg_index + 1]
-    assert actual_bucket_name == expected_bucket_name, (
-        f"pod_spec container args have unexpected default-artifact-root."
-        f"  Expected {expected_bucket_name}, found {actual_bucket_name}"
-    )
-
-    # test correct data structure is sent to admission webhook
-    mlflow_pod_defaults_data = {
-        key.name: value
-        for key, value in harness.model.get_relation(
-            pod_defaults_rel_name, pod_defaults_rel_id
-        ).data.items()
-        if "mlflow-server" in key.name
-    }
-    mlflow_pod_defaults_minio_data = json.loads(
-        mlflow_pod_defaults_data[charm_name]["pod-defaults"]
-    )["minio"]["env"]
-
-    assert (
-        mlflow_pod_defaults_minio_data["MLFLOW_S3_ENDPOINT_URL"]
-        == f"http://{os_data_dict['service']}.{os_data_dict['namespace']}:{os_data_dict['port']}"
-    )
-    assert (
-        mlflow_pod_defaults_minio_data["MLFLOW_TRACKING_URI"]
-        == f"http://{harness.model.app.name}.{harness.model.name}.svc.cluster.local:{harness.charm.config['mlflow_port']}"  # noqa: E501
+    @patch(
+        "charm.KubernetesServicePatch",
+        lambda x, y, service_name, service_type, refresh_event: None,
     )
+    def test_get_object_storage_data_success(self, obejct_storage_relation: Harness):
+        obejct_storage_relation.begin_with_initial_hooks()
+        assert obejct_storage_relation.charm.model.unit.status == WaitingStatus(
+            "Waiting for relational-db relation data"
+        )
 
+    @patch(
+        "charm.KubernetesServicePatch",
+        lambda x, y, service_name, service_type, refresh_event: None,
+    )
+    @patch("charm.MlflowCharm._validate_default_s3_bucket_name_and_access")
+    def test_get_relational_db_data_success(
+        self, validate_default_s3_bucket: MagicMock, relational_db_relation: Harness
+    ):
+        relational_db_relation.begin_with_initial_hooks()
```

Comment:
Can we clarify the nomenclature here?  `relational_db_relation` is really a `harness` that has db and object storage relations, but that doesn't come across here.